File: c28.c
#include <stdio.h>

int main() {
    int a = 10, b = 3;
    float x = 5.5, y = 2.0;
    
    // Arithmetic operators
    printf("=== Arithmetic Operators ===\n");
    printf("%d + %d = %d\n", a, b, a + b);
    printf("%d - %d = %d\n", a, b, a - b);
    printf("%d * %d = %d\n", a, b, a * b);
    printf("%d / %d = %d\n", a, b, a / b);
    printf("%d %% %d = %d\n", a, b, a % b);
    
    // Relational operators
    printf("\n=== Relational Operators ===\n");
    printf("%d > %d: %d\n", a, b, a > b);
    printf("%d < %d: %d\n", a, b, a < b);
    printf("%d == %d: %d\n", a, b, a == b);
    printf("%d != %d: %d\n", a, b, a != b);
    
    // Logical operators
    printf("\n=== Logical Operators ===\n");
    printf("(%d > 5) && (%d < 5): %d\n", a, b, (a > 5) && (b < 5));
    printf("(%d > 5) || (%d > 5): %d\n", a, b, (a > 5) || (b > 5));
    printf("!(%d > 5): %d\n", a, !(a > 5));
    
    // Assignment operators
    printf("\n=== Assignment Operators ===\n");
    int c = a;
    printf("c = a: %d\n", c);
    c += 5;
    printf("c += 5: %d\n", c);
    
    // Increment and Decrement
    printf("\n=== Increment/Decrement ===\n");
    printf("a++ = %d, a after = %d\n", a++, a);
    printf("++b = %d, b after = %d\n", ++b, b);
    
    return 0;
}

File: level 1/c28.cs
using System;

class OperatorDemo
{
    static void Main()
    {
        // Arithmetic Operators
        int a = 10, b = 3;
        Console.WriteLine("=== Arithmetic Operators ===");
        Console.WriteLine($"a + b = {a + b}");
        Console.WriteLine($"a - b = {a - b}");
        Console.WriteLine($"a * b = {a * b}");
        Console.WriteLine($"a / b = {a / b}");
        Console.WriteLine($"a % b = {a % b}");

        // Comparison Operators
        Console.WriteLine("\n=== Comparison Operators ===");
        Console.WriteLine($"a == b: {a == b}");
        Console.WriteLine($"a != b: {a != b}");
        Console.WriteLine($"a > b: {a > b}");
        Console.WriteLine($"a < b: {a < b}");
        Console.WriteLine($"a >= b: {a >= b}");
        Console.WriteLine($"a <= b: {a <= b}");

        // Logical Operators
        bool x = true, y = false;
        Console.WriteLine("\n=== Logical Operators ===");
        Console.WriteLine($"x && y: {x && y}");
        Console.WriteLine($"x || y: {x || y}");
        Console.WriteLine($"!x: {!x}");

        // Assignment Operators
        Console.WriteLine("\n=== Assignment Operators ===");
        int c = 5;
        c += 3;
        Console.WriteLine($"After c += 3: {c}");
        c -= 2;
        Console.WriteLine($"After c -= 2: {c}");
        c *= 2;
        Console.WriteLine($"After c *= 2: {c}");

        // Bitwise Operators
        Console.WriteLine("\n=== Bitwise Operators ===");
        Console.WriteLine($"a & b: {a & b}");
        Console.WriteLine($"a | b: {a | b}");
        Console.WriteLine($"a ^ b: {a ^ b}");
        Console.WriteLine($"~a: {~a}");
    }
}